sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubServiceClient.cs

```diff
@@ -0,0 +1,969 @@
+            this.hub = hub;
+            _credential = credential;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebPubSubServiceClient"/>.
+        /// </summary>
+        /// <param name="connectionString">Connection string contains Endpoint and AccessKey.</param>
+        /// <param name="hub">Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore.</param>
+        public WebPubSubServiceClient(string connectionString, string hub): this(ParseConnectionString(connectionString), hub)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebPubSubServiceClient"/>.
+        /// </summary>
+        /// <param name="connectionString">Connection string contains Endpoint and AccessKey.</param>
+        /// <param name="hub">Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore.</param>
+        /// <param name="options"></param>
+        public WebPubSubServiceClient(string connectionString, string hub, WebPubSubServiceClientOptions options) : this(ParseConnectionString(connectionString), hub, options)
+        {
+        }
+
+        /// <summary>
+        /// This constructor is intended to be used for mocking.
+        /// </summary>
+        protected WebPubSubServiceClient()
+        {
+        }
+
+        internal WebPubSubServiceClient(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri vaultBaseUrl)
+        {
+            RestClient = new WebPubSubRestClient(clientDiagnostics, pipeline, vaultBaseUrl, ApiVersion);
+            _clientDiagnostics = clientDiagnostics;
+        }
+
+        private WebPubSubServiceClient((Uri Endpoint, AzureKeyCredential Credential) parsedConnectionString, string hub) : this(
+            parsedConnectionString.Endpoint, hub, parsedConnectionString.Credential)
+        {
+        }
+
+        private WebPubSubServiceClient((Uri Endpoint, AzureKeyCredential Credential) parsedConnectionString, string hub, WebPubSubServiceClientOptions options) : this(
+            parsedConnectionString.Endpoint, hub, parsedConnectionString.Credential, options)
+        {
+        }
+
+        /// <summary>Broadcast content inside request body to all the connected client connections.</summary>
+        /// <param name="message"></param>
+        /// <param name="excluded">Excluded connection Ids.</param>
+        /// <param name="cancellationToken">The cancellation token to use.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        public virtual async Task<Response> SendToAllAsync(string message, IEnumerable<string> excluded = null, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("WebPubSubServiceRestClient.BroadcastAsync");
```

Comment:
Nit - use nameofs for all these so they don't get out of date as soon as you refactor...  like both parts of this name.
```suggestion
            using var scope = _clientDiagnostics.CreateScope($"{nameof(WebPubSubServiceClient)}.{nameof(SendToAll)}");
```